### ISSUE TITLE
[Bug]: UI duplicate itself

### ISSUE THREAD

**Issue by shaniqwa-drm** [storybookjs/storybook#26810]
### Describe the bug

It's kinda hard to explain but sometimes one photo is worth a thousand words: 
![Screenshot 2024-04-11 at 12 13 14](https://github.com/storybookjs/storybook/assets/124711028/4ca1caea-94cf-4fc6-8579-4b2be7dd083e)

I think it has something to do with HMR, when I refresh the page the bug disappears but after a few moments of work and going between tabs, I come back to Storybook and have to refresh again.

also there is a console error:  `manager  received preloadStories but was unable to determine the source of the event`

### To Reproduce

sorry I have no idea how to reproduce it. I think it started after updating a version

### System

```bash
Storybook Environment Info:
(node:11179) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

  System:
    OS: macOS 14.4.1
    CPU: (8) arm64 Apple M2
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.2.0 - /opt/homebrew/bin/node
    Yarn: 4.0.2 - /opt/homebrew/bin/yarn <----- active
    npm: 10.2.3 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 123.0.6312.123
    Safari: 17.4.1
  npmPackages:
    @storybook/addon-actions: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/addon-docs: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/addon-essentials: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/addon-interactions: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/addon-links: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/addon-mdx-gfm: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/blocks: ^8.0.5 => 8.0.5 
    @storybook/cli: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/test: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/test-runner: ^0.16.0 => 0.16.0 
    @storybook/vue3: ^8.0.0-rc.1 => 8.0.0-rc.1 
    @storybook/vue3-vite: ^8.0.0-rc.1 => 8.0.0-rc.1 
    eslint-plugin-storybook: ^0.6.15 => 0.6.15 
    storybook: ^8.0.0-rc.1 => 8.0.0-rc.1 
    storybook-tailwind-foundations: ^1.1.2 => 1.1.2 
    storybook-vue3-router: ^5.0.0 => 5.0.0
```

**Comment by valentinpalkovic** [comment]
Hi @shaniqwa-drm 

Thank you for reporting this very weird error. I can see that you are on an 8.0 RC version. Could you run `npx storybook@latest upgrade` to upgrade to the latest stable 8.0 version of Storybook?

**Comment by shaniqwa-drm** [comment]
@valentinpalkovic sure:
```
Storybook Environment Info:
(node:21881) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

  System:
    OS: macOS 14.4.1
    CPU: (8) arm64 Apple M2
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.2.0 - /opt/homebrew/bin/node
    Yarn: 4.0.2 - /opt/homebrew/bin/yarn <----- active
    npm: 10.2.3 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 123.0.6312.123
    Safari: 17.4.1
  npmPackages:
    @storybook/addon-actions: ^8.0.6 => 8.0.7 
    @storybook/addon-controls: ^8.0.7 => 8.0.7 
    @storybook/addon-docs: ^8.0.6 => 8.0.7 
    @storybook/addon-essentials: ^8.0.6 => 8.0.7 
    @storybook/addon-interactions: ^8.0.6 => 8.0.7 
    @storybook/addon-links: ^8.0.6 => 8.0.7 
    @storybook/addon-mdx-gfm: ^8.0.6 => 8.0.7 
    @storybook/blocks: ^8.0.6 => 8.0.7 
    @storybook/cli: ^8.0.6 => 8.0.7 
    @storybook/test: ^8.0.6 => 8.0.7 
    @storybook/test-runner: ^0.17.0 => 0.17.0 
    @storybook/vue3: ^8.0.6 => 8.0.7 
    @storybook/vue3-vite: ^8.0.6 => 8.0.7 
    eslint-plugin-storybook: ^0.8.0 => 0.8.0 
    storybook: ^8.0.6 => 8.0.7 
    storybook-tailwind-foundations: ^1.1.2 => 1.1.2 
    storybook-vue3-router: ^5.0.0 => 5.0.0 

```

**Comment by shaniqwa-drm** [comment]
I have somehow managed to find how to fix it.. I am using https://storybook.js.org/addons/storybook-vue3-router and I configured mock routes decorator in my `preview.js` file. I suspected the bug has something to do with decorators, so I removed them one by one until I saw that removing the mock routes decorator solves the issue! I moved it to the several components who need it:
```
Primary.decorators = [
  vueRouter(mockRoutes, {
    vueRouterOptions: {
      linkActiveClass: 'activeLink',
      routes: mockRoutes,
      history: createWebHistory(import.meta.env.BASE_URL),
    },
  }),
]
```
and deleted it from `preview.js`.
